R/crew_launcher_lsf.R

```diff
@@ -0,0 +1,287 @@
+#'   LSF workers.
+#' @export
+#' @family launchers
+#' @description Create an `R6` object to launch and maintain
+#'   workers as LSF jobs.
+#' @details WARNING: the `crew.cluster` LSF plugin is experimental
+#'   and has not actually been tested on a LSF cluster. Please proceed
+#'   with caution and report bugs to
+#'   <https://github.com/wlandau/crew.cluster>.
+#'
+#'   To launch a LSF worker, this launcher
+#'   creates a temporary job script with a call to `crew::crew_worker()`
+#'   and submits it as an LSF job with `sbatch`. To see most of the lines
+#'   of the job script in advance, use the `script()` method of the launcher.
+#'   It has all the lines except for the job name and the
+#'   call to `crew::crew_worker()`, both of
+#'   which will be inserted at the last minute when it is time
+#'   to actually launch a worker.
+#' @inheritParams crew_launcher_cluster
+#' @param lsf_log_output Character of length 1, file pattern to control
+#'   the locations of the LSF worker log files. By default, both standard
+#'   output and standard error go to the same file.
+#'   `lsf_log_output = "crew_log_%J.log"` translates to a line of
+#'   `#BSUB-o crew_log_%J.log` in the LSF job script,
+#'   where `%J` is replaced by the job ID of the worker.
+#'   The default is `/dev/null` to omit these logs.
+#'   Set `lsf_log_output = NULL` to omit this line from the job script.
+#' @param lsf_log_error Character of length 1, file pattern for standard
+#'   error. `lsf_log_error = "crew_error_%J.err"` translates to a line of
+#'   `#BSUB-e crew_error_%J.err` in the LSF job script,
+#'   where `%J` is replaced by the job ID of the worker.
+#'   The default is `/dev/null` to omit these logs.
+#'   Set `lsf_log_error = NULL` to omit this line from the job script.
+#' @param lsf_memory_megabytes_per_cpu Positive numeric of length 1
+#'   with the megabytes of memory required.
+#'   `lsf_memory_megabytes_per_cpu = 4096`
+#'   translates to a line of `#BSUB-M 4096`
+#'   in the LSF job script.
+#'   `lsf_memory_megabytes_per_cpu = NULL` omits this line.
+#' @param lsf_cpus_per_task Optional positive integer of length 1,
+#'   number of CPUs for the worker.
+#'   `lsf_cpus_per_task = 4` translates
+#'   to a line of `#BSUB-n 4` in the LSF job script.
+#'   `lsf_cpus_per_task = NULL` omits this line.
+crew_launcher_lsf <- function(
+  name = NULL,
+  seconds_launch = 60,
+  seconds_interval = 0.001,
+  seconds_timeout = 10,
+  seconds_idle = Inf,
+  seconds_wall = Inf,
+  seconds_exit = 1,
+  tasks_max = Inf,
+  tasks_timers = 0L,
+  reset_globals = TRUE,
+  reset_packages = FALSE,
+  reset_options = FALSE,
+  garbage_collection = FALSE,
+  verbose = FALSE,
+  command_submit = as.character(Sys.which("bsub")),
+  command_delete = as.character(Sys.which("bkill")),
+  script_directory = tempdir(),
+  script_lines = character(0L),
+  lsf_log_output = "/dev/null",
+  lsf_log_error = "/dev/null",
+  lsf_memory_megabytes_per_cpu = NULL,
+  lsf_cpus_per_task = NULL
```

Comment:
I see there is a `-cwd` flag in LSF. I think this will be extremely important for e.g. `targets` workflows, and it would be great to have an `lsf_cwd` argument with default `TRUE`, similar to `sge_cwd` in the SGE launcher.